### PR TITLE
Question and answer

### PR DESCRIPTION
**Pull request by daniel-he-dev**
completed all component conditional rendering
-more questions button
-more/collapse answers button
hover styling for clickable elements
toggle appearance of helpful 'Yes' tags and 'Report' button when clicked
completed search filtering functionality for questions

@cayla-c @jfleming9357 @mattBman23

### CODE REVIEW

File: client/components/QuestionsAndAnswers/Answer.jsx [hunk 1]
```diff
@@ -1,18 +1,43 @@
...
+          onClick={handleToggleHelpful}
+          style={helpful ? {textDecoration: 'none'} : null}
+        >Yes</span>
+        {` (${answer.helpfulness})  | `}
+        {reported
+          ? <span
+              className="d-underlined"
+              onClick={() => setReported(false)}
+              style={{filter: "brightness(300%)", textDecoration: 'none'}}
+            >Reported</span>
```
Comment (jfleming9357) [inline client/components/QuestionsAndAnswers/Answer.jsx:32]: Should this be clickable once reported=true? From the business requirements: 
'Clicking on this link will mark the answer for internal review.  A user should not be able to report an answer more than once.   After clicking on this link, the “Report” link should change to static text that reads “Reported”.   Answers that have been reported should be marked as such in the system for further action to be taken.'

**Review by jfleming9357 — APPROVED** [review]
Other than that one comment, looks good! Looks like it should work fine so I'm good with merging as is.

File: client/components/QuestionsAndAnswers/Answer.jsx (hunk 1 above)
Comment (daniel-he-dev) [reply client/components/QuestionsAndAnswers/Answer.jsx:31]: Good catch, fixed this.

**Comment by mattBman23** [comment]
Look good

**Pull request merged**